rclcpp/include/rclcpp/logger.hpp

```diff
@@ -161,6 +162,10 @@ class Logger
   RCLCPP_PUBLIC
   void
   set_level(Level level);
+
+  RCLCPP_PUBLIC
+  rcpputils::fs::path
+  get_logging_directory() const;
```

Comment:
What's the thinking behind making this a class method rather than a free function or a static class method?

Reply:
I was going to make it a static class method, but maybe we wanted in the future to support different loggers to use different directories.
If we're never going to support that, a static method sounds better.

Reply:
Modified to a `static` method in https://github.com/ros2/rclcpp/pull/1509/commits/1b6c60f52776d92f7966bd4210138cea76067333.

Reply:
Well... why not a free function? I guess I'm trying to understand why this is the right namespace for it. It doesn't exactly map to the rcl function's style. That would be something more like `rclcpp::Logging::get_logging_directory()`, which I am not advocating for.

I actually think a free function would make more sense. Sorry for the back and forth.

Reply:
Sounds good to me, see https://github.com/ros2/rclcpp/pull/1509/commits/3c42fa6c3485a7cdfd72969b7757c0a29c1b065d.

The new function is defined in `rclcpp/logger.hpp`, but you can also include [rclcpp/logging.hpp](https://github.com/ros2/rclcpp/blob/master/rclcpp/resource/logging.hpp.em) and it will be available.